packages/vite/src/node/optimizer/scan.ts

```diff
@@ -494,6 +494,16 @@ function esbuildScanPlugin(
           config.optimizeDeps?.esbuildOptions?.loader?.[`.${ext}`] ||
           (ext as Loader)
 
+        if (contents.includes('import.meta.glob')) {
```

Comment:
I think we need to transform TS into JS before passing to `transformGlobImport` like here. Extracting this part into a function would be nice.
https://github.com/vitejs/vite/blob/2f7d1c1f630f9dbbd44ce235e86afc09e4b9cc85/packages/vite/src/node/optimizer/scan.ts#L307-L331